service/matching/matching_engine.go

```diff
@@ -617,9 +617,19 @@ pollLoop:
 		if err != nil {
 			switch err.(type) {
 			case *serviceerror.Internal, *serviceerror.DataLoss:
-				e.nonRetryableErrorsDropTask(task, taskQueueName, err)
-				// drop the task as otherwise task would be stuck in a retry-loop
-				task.finish(nil, false)
+				if e.config.MatchingDropNonRetryableTasks() {
+					e.nonRetryableErrorsDropTask(task, taskQueueName, err)
+					// drop the task as otherwise task would be stuck in a retry-loop
+					task.finish(nil, false)
+				} else {
+					// default case
+					task.finish(err, false)
+					if err.Error() == common.ErrNamespaceHandover.Error() {
```

Comment:
ErrNamespaceHandover is unavailable error right? when will this condition be true?

Reply:
ah, yes you are indeed right. Apologies for not catching this - fixed this.